compiler/rustc_target/src/target_features.rs

```diff
@@ -407,6 +407,79 @@ const IBMZ_ALLOWED_FEATURES: &[(&str, Stability)] = &[
     // tidy-alphabetical-end
 ];
 
+const X86_IMPLIED_FEATURES: &[(&str, &[&str])] = &[
```

Comment:
Is it necessary to have this list here?

So far, we handle implied target features by  having LLVM compute the full list of target features. That way, `-C target-feature=avx2` makes `cfg!(target_feature="avx")` be true -- at least that is my understanding.

We shouldn't have two different implied-target-feature mechanisms: either everything should use LLVM, or everything should use this table.

Reply:
There seemed to be some desire in zulip to keep it separate, but I can do that (if I can find where `cfg` is set from LLVM...)

Reply:
Where on Zulip is that desire expressed? All I can see i Amanieu saying we should have our own implication table, but I don't see anyone even talking about the concern that different parts of rustc would have different user-visible notions of "implied target features".

Reply:
I think that we should handle implied target features entirely on the rustc side and only use LLVM to query target features when `-C target-cpu` is used (at least until we have our own mechanism for that). I want to move away from LLVM's target feature and into something that is backend-agnostic.